libmscore/album.cpp

```diff
@@ -0,0 +1,1489 @@
+        box->setOffset(0, 0);
+        box->setBoxHeight(Spatium(pageHeight * 0.05 / scoreSpatium));
+    }
+
+    // make sure that we have these 3 text fields
+    MeasureBase* measure = getCombinedScore()->measures()->first();
+    measure->clearElements();
+    Text* s = new Text(getCombinedScore(), Tid::TITLE);
+    s->setPlainText("");
+    s->setSize(36);
+    measure->add(s);
+    s = new Text(getCombinedScore(), Tid::COMPOSER);
+    s->setPlainText("");
+    s->setSize(16);
+    measure->add(s);
+    s = new Text(getCombinedScore(), Tid::POET);
+    s->setPlainText("");
+    s->setSize(16);
+    measure->add(s);
+
+    for (auto& x : getCombinedScore()->measures()->first()->el()) {
+        if (x && x->isText()) {
+            Text* t = toText(x);
+
+            if (t->tid() == Tid::TITLE) {
+                t->setPlainText(albumTitle());
+            } else if (t->tid() == Tid::COMPOSER) {
+                if (m_drawFrontCover) {
+                    t->setPlainText(composers().join("\n"));
+                }
+            } else if (t->tid() == Tid::POET) {
+                if (m_drawFrontCover) {
+                    t->setPlainText(lyricists().join("\n"));
+                }
+            }
+        }
+    }
+}
+
+//---------------------------------------------------------
+//   updateContents
+///     Updates (or creates or deletes) the movement and content responsible
+///     for the Contents pages.
+//---------------------------------------------------------
+
+void Album::updateContents()
+{
+    if (!getCombinedScore()) {
+        return;
+    }
+
+    //
+    // delete the contents pages-movement
+    //
+    if (!generateContents()) {
+        if (m_combinedScore->movements()->at(1)->textMovement()) {
+            MasterScore* contentsMovement = m_combinedScore->movements()->at(1);
+            m_combinedScore->removeMovement(1);
+            delete contentsMovement;
+        }
+        return;
+    }
+
+    qreal pageWidth = getCombinedScore()->pages().at(0)->width();
+    qreal pageHeight = getCombinedScore()->pages().at(0)->height();
+    qreal scoreSpatium = getCombinedScore()->spatium();
+    int charWidth = pageWidth / scoreSpatium;
+
+    //
+    // if there is no contents movement create one
+    //
+    if (!getCombinedScore()->movements()->at(1)->textMovement()) {
+        MasterScore* ms = m_albumItems.at(0)->score()->clone();
+        ms->setName("Contents");
+        ms->setTextMovement(true);
+        getCombinedScore()->insertMovement(ms, 1);
+
+        while (ms->systems().size() > 1) {
+            for (auto& x : ms->systems().last()->measures()) {
+                ms->removeElement(x);
+            }
+            ms->systems().removeLast();
+        }
+
+        // make sure that we have these 2 text fields
+        MeasureBase* measure = ms->measures()->first();
+        measure->clearElements();
+        Text* s = new Text(ms, Tid::TITLE);
+        s->setPlainText("Contents");
+        s->setSize(36);
```

Comment:
Why this magic number? (and others in the vicinity)

Reply:
I'd rather the the style size of a Title text here. Or that size plus some magic number ;-)

Reply:
I believe that it would be best to create new styles for the front cover and the contents pages.